hadoop-ozone/recon/src/main/java/org/apache/hadoop/ozone/recon/persistence/ContainerHealthSchemaManager.java

```diff
@@ -123,7 +126,26 @@ public void insertUnhealthyContainerRecords(List<UnhealthyContainers> recs) {
             rec.getContainerState());
       });
     }
-    unhealthyContainersDao.insert(recs);
+
+    try (Connection connection = containerSchemaDefinition.getDataSource().getConnection()) {
+      connection.setAutoCommit(false); // Turn off auto-commit for transactional control
+      for (UnhealthyContainers rec : recs) {
```

Comment:
The patch logs the error inside the loop for each failed insert, which could lead to excessive logging if a large number of records fail ?

Reply:
changed to `debug`